src/Drupal/ConfigTrait.php

```
@@ -0,0 +1,478 @@
+   * before the first write so that revert can delete objects created by the
+   * scenario rather than leaving empty shells behind.
+   *
+   * @var array<string, array{existed: bool, data: array<int|string, mixed>}>
+   */
+  protected array $configOriginalData = [];
+
+  /**
+   * Reset the snapshot registry before each scenario.
+   */
+  #[BeforeScenario('@api')]
+  public function configBeforeScenario(BeforeScenarioScope $scope): void {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$scope'. (undefined)

(UnusedFormalParameter)